configure.self

```diff
@@ -217,6 +217,15 @@ const char *foo (void) {
 void foo (void) {
   pthread_setname_np(pthread_self(), "abc");
 }
+'
```

Comment:
If SETNAME_GNU works we shouldn't check for DARWIN, so change this to
`' || \`

Reply:
Done.